### PR TITLE
Implemented the Knuth Morris Pattern Matching Algorithm.

### PR DESCRIPTION
**Pull request by srimani-programmer**
Fixes #2555 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implemented the Knuth Morris Pattern Matching Algorithm in PHP

#### Languages Used:
- PHP

#### Files Added:
- KMP.php

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Knuth_Morris_Pratt_Algorithm/KMP.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            } else { 
+            $lps[$i] = 0; 
+            $i++; 
+            } 
+        } 
+    } 
+}
+
+// A function to perform the Knuth Morris Algorithm for string matching.
+
```
Comment (asha15) [inline Knuth_Morris_Pratt_Algorithm/KMP.php:28]: remove this line

File: Knuth_Morris_Pratt_Algorithm/KMP.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    OUTPUT:
+
+    Pattern found at index : 0
+    Pattern found at index : 9
+    Pattern found at index : 16
+  */
+?>
+
+
+
```
Comment (asha15) [inline Knuth_Morris_Pratt_Algorithm/KMP.php:91]: remove these lines

**Comment by asha15** [comment]
please squash your commits , we are good to merge

**Comment by srimani-programmer** [comment]
I need your help. Accidentally I was pushed my "Bucket_Sort" code into this branch. I referred all the docs to resolve this undoing the commit. But I am not able to do it. Can you please help me to undo this Commit and Merge the Code into KMP. Thanks in advance.

**Comment by asha15** [comment]
please reach me on slack, I'll help you

**Comment by asha15** [comment]
please squash the commits

**Comment by asha15** [comment]
it has conflicts :( please resolve it

**Pull request closed without merging**